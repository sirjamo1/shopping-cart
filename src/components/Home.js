import React from "react";

const Home = () => {
    const text = "Chartreuse";
    return (
        <div className="home-container">
            <h1 className="home-title" data-testid="homeTitle">{text}</h1>
            <div className="home-banner"></div>
            <div className="general-info">
                <h4>What is Chartreuse</h4>
                <p>
                    Chartreuse is a unique and well-known liqueur that has a
                    rich history dating back to the 18th century. The recipe for
                    Chartreuse was originally created by a group of Carthusian
                    monks in the French Alps. The recipe was said to have been
                    passed down to the monks by a marshal of France, François
                    Annibal d'Estrées, in the early 1700s. The recipe consisted
                    of a secret blend of over 130 herbs, plants, and flowers,
                    and it was said to have medicinal properties.
                </p>
            </div>
        </div>
    );
};



export default Home