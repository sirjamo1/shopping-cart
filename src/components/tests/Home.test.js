import Home from "../Home";
import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

afterEach(cleanup);

it("renders", () => {
    const { asFragment } = render(<Home text="Chartreuse" />);
    expect(asFragment()).toMatchSnapshot();
});

it("inserts text in h1", () => {
    const { getByTestId, getByText } = render(<Home text="Chartreuse" />)
    expect(getByTestId("homeTitle")).toHaveTextContent("Chartreuse");
    expect(getByText("Chartreuse")).toHaveClass("home-title");
});
