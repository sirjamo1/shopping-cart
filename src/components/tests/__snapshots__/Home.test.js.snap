// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="home-container"
  >
    <h1
      class="home-title"
      data-testid="homeTitle"
    >
      Chartreuse
    </h1>
    <div
      class="home-banner"
    />
    <div
      class="general-info"
    >
      <h4>
        What is Chartreuse
      </h4>
      <p>
        Chartreuse is a unique and well-known liqueur that has a rich history dating back to the 18th century. The recipe for Chartreuse was originally created by a group of Carthusian monks in the French Alps. The recipe was said to have been passed down to the monks by a marshal of France, François Annibal d'Estrées, in the early 1700s. The recipe consisted of a secret blend of over 130 herbs, plants, and flowers, and it was said to have medicinal properties.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
